stdlib/ssl.pyi

```diff
@@ -176,39 +199,72 @@ OPENSSL_VERSION: str
+class AlertDescription(enum.IntEnum):
+    ALERT_DESCRIPTION_ACCESS_DENIED: int
+    ALERT_DESCRIPTION_BAD_CERTIFICATE: int
+    ALERT_DESCRIPTION_BAD_CERTIFICATE_HASH_VALUE: int
+    ALERT_DESCRIPTION_BAD_CERTIFICATE_STATUS_RESPONSE: int
+    ALERT_DESCRIPTION_BAD_RECORD_MAC: int
+    ALERT_DESCRIPTION_CERTIFICATE_EXPIRED: int
+    ALERT_DESCRIPTION_CERTIFICATE_REVOKED: int
+    ALERT_DESCRIPTION_CERTIFICATE_UNKNOWN: int
+    ALERT_DESCRIPTION_CERTIFICATE_UNOBTAINABLE: int
+    ALERT_DESCRIPTION_CLOSE_NOTIFY: int
+    ALERT_DESCRIPTION_DECODE_ERROR: int
+    ALERT_DESCRIPTION_DECOMPRESSION_FAILURE: int
+    ALERT_DESCRIPTION_DECRYPT_ERROR: int
+    ALERT_DESCRIPTION_HANDSHAKE_FAILURE: int
+    ALERT_DESCRIPTION_ILLEGAL_PARAMETER: int
+    ALERT_DESCRIPTION_INSUFFICIENT_SECURITY: int
+    ALERT_DESCRIPTION_INTERNAL_ERROR: int
+    ALERT_DESCRIPTION_NO_RENEGOTIATION: int
+    ALERT_DESCRIPTION_PROTOCOL_VERSION: int
+    ALERT_DESCRIPTION_RECORD_OVERFLOW: int
+    ALERT_DESCRIPTION_UNEXPECTED_MESSAGE: int
+    ALERT_DESCRIPTION_UNKNOWN_CA: int
+    ALERT_DESCRIPTION_UNKNOWN_PSK_IDENTITY: int
+    ALERT_DESCRIPTION_UNRECOGNIZED_NAME: int
+    ALERT_DESCRIPTION_UNSUPPORTED_CERTIFICATE: int
+    ALERT_DESCRIPTION_UNSUPPORTED_EXTENSION: int
+    ALERT_DESCRIPTION_USER_CANCELLED: int
+
+ALERT_DESCRIPTION_HANDSHAKE_FAILURE: AlertDescription
+ALERT_DESCRIPTION_INTERNAL_ERROR: AlertDescription
+ALERT_DESCRIPTION_ACCESS_DENIED: AlertDescription
+ALERT_DESCRIPTION_BAD_CERTIFICATE: AlertDescription
+ALERT_DESCRIPTION_BAD_CERTIFICATE_HASH_VALUE: AlertDescription
+ALERT_DESCRIPTION_BAD_CERTIFICATE_STATUS_RESPONSE: AlertDescription
+ALERT_DESCRIPTION_BAD_RECORD_MAC: AlertDescription
+ALERT_DESCRIPTION_CERTIFICATE_EXPIRED: AlertDescription
+ALERT_DESCRIPTION_CERTIFICATE_REVOKED: AlertDescription
+ALERT_DESCRIPTION_CERTIFICATE_UNKNOWN: AlertDescription
+ALERT_DESCRIPTION_CERTIFICATE_UNOBTAINABLE: AlertDescription
+ALERT_DESCRIPTION_CLOSE_NOTIFY: AlertDescription
+ALERT_DESCRIPTION_DECODE_ERROR: AlertDescription
+ALERT_DESCRIPTION_DECOMPRESSION_FAILURE: AlertDescription
+ALERT_DESCRIPTION_DECRYPT_ERROR: AlertDescription
+ALERT_DESCRIPTION_ILLEGAL_PARAMETER: AlertDescription
+ALERT_DESCRIPTION_INSUFFICIENT_SECURITY: AlertDescription
+ALERT_DESCRIPTION_NO_RENEGOTIATION: AlertDescription
+ALERT_DESCRIPTION_PROTOCOL_VERSION: AlertDescription
+ALERT_DESCRIPTION_RECORD_OVERFLOW: AlertDescription
+ALERT_DESCRIPTION_UNEXPECTED_MESSAGE: AlertDescription
+ALERT_DESCRIPTION_UNKNOWN_CA: AlertDescription
+ALERT_DESCRIPTION_UNKNOWN_PSK_IDENTITY: AlertDescription
+ALERT_DESCRIPTION_UNRECOGNIZED_NAME: AlertDescription
+ALERT_DESCRIPTION_UNSUPPORTED_CERTIFICATE: AlertDescription
+ALERT_DESCRIPTION_UNSUPPORTED_EXTENSION: AlertDescription
+ALERT_DESCRIPTION_USER_CANCELLED: AlertDescription
 
 class _ASN1Object(NamedTuple):
     nid: int
     shortname: str
     longname: str
     oid: str
+    @classmethod
+    def fromnid(cls, nid: int) -> _ASN1Object: ...
```

Comment:
Strictly these should have `cls: Type[Self]` and `-> Self` because they'll return a subclass if you subclass this object. (Where Self is from `from _typeshed import Self`)

Reply:
that sounds interesting and makes sense, will change it